cyral/resource_cyral_repository.go

```diff
@@ -409,9 +409,20 @@ func resourceRepository() *schema.Resource {
 							Optional:    true,
 						},
 						RepoNodeDynamicKey: {
-							Description: "Indicates if node is dynamically discovered. If true, `host` and `port` must be empty.",
-							Type:        schema.TypeBool,
-							Optional:    true,
+							Description: "*Only supported for MongoDB in cluster configurations.*\n" +
+								"Indicates if the node is dynamically discovered, meaning that the sidecar " +
+								"will query the cluster to get the topology information and discover the " +
+								"hosts of the remaining nodes. If set to `true`, `host` and `port` must " +
+								"be empty. A node that does not declare this field is considered `static`.\n" +
+								"The following conditions apply: \n" +
+								"  - The total number of declared `" + RepoNodesKey + "` blocks must match " +
+								"the actual number of nodes in the cluster.\n" +
+								"  - If there are static nodes in the configuration, they must be declared " +
+								"before all dynamic nodes.\n" +
+								"  - See the MongoDB-specific configuration in the [" + RepoMongoDBSettingsKey +
+								"](#nested-schema-for-" + RepoMongoDBSettingsKey + ").",
+							Type:     schema.TypeBool,
```

Comment:
Is it possible (and desirable) to specify default value (since the parameter is optional)?

Reply:
AFAIK this `TypeBool` is mapped to an underlying `bool` not a `*bool`, so it should not allow nulls. In this sense, I understand that the default here should be `false` then. If that is the case, then this should be the proper value for the default correct? I will double check this, though.

Reply:
Having second thoughts on my previous argument and I see now it does not make sense from the user standpoint. We should indeed provide the default values in the code as they would reflect in the docs and give a better notion to our users what is the behaviour when a given argument is not provided. Anyway, I think we should address these defaults in a different PR as this would require some tests. @ricardorey10 , could you take a stab on providing these default values to the schema?

Reply:
Sure thing. I'll put my thoughts here for discussion, and we can create a PR later (I'm a bit tied up with some Snowflake stuff now so I can't volunteer to do that change right now, but if it can wait a few days I can take it):

- "host": this should default to an empty string, as we cannot guess anything better.
- "port": this should default to zero, as we cannot guess anything.
- "dynamic": I think the default should be `false`. The reasoning here is that, if I supply the host and port but I don't supply `dynamic`, then this will result in an API error, because we can't have `dynamic = true` and non-empty host:port. This will also make it obvious that, for dynamic repo nodes, the *only* thing they have to configure is the dynamic-ness os the node (as `true`).

With the above defaults, we can have the following configurations:

```
repo_node {
  dynamic = true   // Host and port are empty, which is fine
}
```

```
repo_node {
  host = "something"    // dynamic is false, which is what we want
  port = 1234
}
```

```
repo_node {
  host = "something"    // Just combining the two above in a single config
  port = 1234
}

repo_node {
  dynamic = true
}
```